### PR TITLE
Add "plantuml_markdown" plugin.

### PR DESCRIPTION
**Pull request by matthew16550**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -48,6 +48,13 @@ jobs:
           PLANTUML_EXEC: java -Djava.awt.headless=true -jar ${{github.workspace}}/plantuml.jar
         run: |
           py.test --color=yes tests/
+      - name: Archive failures
+        uses: actions/upload-artifact@v2
```
Comment (Kwpolska) [inline .github/workflows/ci.yml:52]: I’d prefer not to, just `cat` those files in the build script, or ensure enough of them is shown by pytest.

File: tests/__init__.py [hunk 2]
```diff
@@ -23,3 +35,34 @@ def execute_plugin_tasks(plugin: Task, verbosity: int = 0):
         catched = t.execute(stream)
         if catched:
             raise Exception("Task error for '{}'\n{}".format(t.name, catched.get_msg()))
+
+
+class regex:  # noqa
```
Comment (Kwpolska) [inline tests/__init__.py:40]: This should be documented and named better. (Also, I’m not 100% sure if we need this.)

File: tests/conftest.py [hunk 3]
```diff
@@ -22,23 +27,77 @@ def f(ext: str, data: str, extra_plugins_dirs: List[Path] = None, metadata: str
...
-
-        return CompileResult(tmp_site_path / 'cache' / 'pages' / 'test.html')
+        post = site.timeline[0]
+        post.compile('en')
+        return CompileResult(request, post)
 
     return f
 
 
+# CompileResult can be used as a Context Manager e.g.
```
Comment (Kwpolska) [inline tests/conftest.py:44]: That should be a docstring in `CompileResult`.

File: tests/test_plantuml_markdown.py [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+    raise pytest.skip("plantuml_markdown plugin requires Python >= 3.6", allow_module_level=True)
+
+from tests import V8_PLUGIN_PATH, regex
+from tests.conftest import CompileResult
+from v8.plantuml_markdown.plantuml_markdown import PlantUmlMarkdownProcessor, first_line_for_listing_block
+
+
+def test_svg(do_fence_test):
+    with do_fence_test('plantuml') as compiled:
+        assert compiled.raw_html == regex(
```
Comment (Kwpolska) [inline tests/test_plantuml_markdown.py:18]: Using regex for HTML is very unclean. Better to parse it with `lxml` and query it. (For example, what if the order of `<text>` elements becomes unstable?

File: v8/plantuml_markdown/README.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+```
+
+```plantuml
+A -> B : This arrow uses the default style
+```
+~~~
+
+# Unicode
+
+The plugin expects Markdown files to be encoded with UTF-8.
```
Comment (Kwpolska) [inline v8/plantuml_markdown/README.md:103]: (So does Nikola, although we tolerate UTF-8 with BOM. You can remove this note if your plugin works with BOMs too — and if it doesn’t, try to fix that.)

File: v8/plantuml_markdown/README.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
+This plugin renders [PlantUML](https://plantuml.com/) in Markdown files.
+
+# Requirements
+
+* [plantuml plugin](https://plugins.getnikola.com/#plantuml) (the actual PlantUML rendering is managed there)
```
Comment (Kwpolska) [inline v8/plantuml_markdown/README.md:5]: Create a `requirements-plugins.txt` file and list it there (as just `plantuml`, [example](https://github.com/getnikola/plugins/blob/master/v8/pkgindex/requirements-plugins.txt))

**Review by Kwpolska — CHANGES_REQUESTED** [review]

File: v8/plantuml_markdown/README.md (hunk 5 above)
Comment (matthew16550) [reply v8/plantuml_markdown/README.md:103]: Yes BOM seems to be fine, `CompileMarkdown.compile()` reads the file and the plugin doesn't know anything about it.

File: tests/__init__.py (hunk 2 above)
Comment (matthew16550) [reply tests/__init__.py:40]: removed

File: tests/test_plantuml_markdown.py (hunk 4 above)
Comment (matthew16550) [reply tests/test_plantuml_markdown.py:18]: You're right, xpaths make it much less brittle.

**Review by Kwpolska — APPROVED** [review]

**Pull request merged**